test/src/main/java/org/apache/accumulo/test/fate/accumulo/FateStoreIT.java

```diff
@@ -216,7 +227,104 @@ protected void testDeferredOverflow(FateStore<TestEnv> store, ServerContext sctx
     } finally {
       executor.shutdownNow();
       // Cleanup so we don't interfere with other tests
-      store.list().forEach(fateIdStatus -> store.reserve(fateIdStatus.getTxid()).delete());
+      // All stores should already be unreserved
+      store.list()
+          .forEach(fateIdStatus -> store.tryReserve(fateIdStatus.getTxid()).orElseThrow().delete());
+    }
+  }
+
+  @Test
+  public void testCreateWithKey() throws Exception {
+    executeTest(this::testCreateWithKey);
+  }
+
+  protected void testCreateWithKey(FateStore<TestEnv> store, ServerContext sctx) {
+    KeyExtent ke1 = new KeyExtent(TableId.of("tableId"), new Text("zzz"), new Text("aaa"));
+    KeyExtent ke2 = new KeyExtent(TableId.of("tableId2"), new Text("zzz"), new Text("aaa"));
+
+    byte[] key1 = serialize(ke1);
+    long tid1 = store.create(key1);
+
+    byte[] key2 = serialize(ke2);
+    long tid2 = store.create(key2);
+    assertNotEquals(tid1, tid2);
+
+    FateTxStore<TestEnv> txStore1 = store.reserve(tid1);
+    FateTxStore<TestEnv> txStore2 = store.reserve(tid2);
+    try {
+      assertTrue(txStore1.timeCreated() > 0);
+      assertEquals(TStatus.NEW, txStore1.getStatus());
+      assertArrayEquals(key1, txStore1.getKey().orElseThrow());
+
+      assertTrue(txStore2.timeCreated() > 0);
+      assertEquals(TStatus.NEW, txStore2.getStatus());
+      assertArrayEquals(key2, txStore2.getKey().orElseThrow());
+
+      assertEquals(2, store.list().count());
+    } finally {
+      txStore1.delete();
+      txStore2.delete();
+    }
```

Comment:
We need to try calling create after delete with the same key and ensure that works.

Reply:
May be better to do this in the in progress test because create when in new status will still succeed.